.github/workflows/integration_test.yml

```diff
@@ -47,22 +52,62 @@ jobs:
         # from https://github.com/Azure/login/commits/master
         uses: azure/login@92a5484dfaf04ca78a94597f4f19fea633851fa2
         with:
-          client-id: ${{ secrets.CD_CLIENT_ID }}
+          client-id: ${{ secrets.CI_CLIENT_ID }}
           tenant-id: ${{ secrets.TENANT_ID }}
           subscription-id: ${{ secrets.SUBSCRIPTION_ID }}
 
+      - name: Delete old reports
+        if: ${{ inputs.allure }}
+        uses: azure/CLI@4db43908b9df2e7ac93c8275a8f9a448c59338dd # v1
+        env:
+          STORAGE_NAME: ${{ vars.INTEGRATION_TEST_STORAGE_ACCOUNT_NAME }}
+          STORAGE_FOLDER: ${{ vars.INTEGRATION_TEST_REPORTS_FOLDER }}
+        with:
+          inlineScript: |
+            az storage blob delete-batch \
+            --source "${{ env.STORAGE_FOLDER }}" \
+            --account-name "${{ env.STORAGE_NAME }}" \
+            --subscription "${{ secrets.SUBSCRIPTION_ID }}" \
+            --auth-mode login
+
       - name: Run Integration Tests
         shell: bash
         run: |
           export SUBKEY=${{ secrets.SUBKEY }}
           export CANARY=${{ inputs.canary }}
+          export TYPE=$([ ${{ inputs.allure }} = true ] && echo "allure" || echo "old")
           export CUCUMBER_PUBLISH_TOKEN=${{ secrets.CUCUMBER_PUBLISH_TOKEN }}
           export ISSUER_RANGE_TABLE=${{ vars.ISSUER_RANGE_TABLE }}
           export AFM_SA_CONNECTION_STRING='${{ secrets.AFM_SA_CONNECTION_STRING }}'
 
           cd ./integration-test
           chmod +x ./run_integration_test.sh
-          ./run_integration_test.sh ${{( github.event.inputs == null && 'dev') || inputs.environment }}
+          ./run_integration_test.sh ${{( github.event.inputs == null && 'dev') || inputs.environment }} $TYPE
+
+      - name: Generate allure report
+        if: ${{ inputs.allure }}
+        shell: bash
+        run: |
```

Comment:
utilizzando docker se falliscono il comando non va in errore, questo vuol dire che non abbiamo mai un fallimento. Il notify alla fine è inutile, dobbiamo andare a controllare il report allure a mano